novice-goals.Rmd

```diff
@@ -0,0 +1,226 @@
+    -   [Anscombe's Quartet and the DataSaurus dozen][anscome-datasaurus]
+-   What are the core elements of every plot?
+    -   data
+    -   geometric objects
+    -   aesthetic mapping
+-   How can I create different kinds of plots?
+    -   scatter plot
+    -   line plot
+    -   histogram
+    -   bar plot
+    -   which to use when
+-   How can I plot multiple datasets at once?
+    -   grouping
+    -   faceting
+-   How can I make misleading plots?
+    -   showing a single central tendency data point instead of the individual observations
+    -   saturated plots instead of for example violins or 2D histograms
+    -   not starting axis at zero to exaggerate a difference
+    -   not using perceptually uniform colormaps to indicate quantities
+    -   not thinking about color blindness
+-   What *isn't* included?
+    -   outliers
+    -   interactive plots
+    -   maps
+    -   3D visualization
+
+## Development
+
+-   How can I make my own functions?
+    -   declaring functions
+    -   declaring parameters
+    -   default values
+    -   common conventions
+-   How can I make my programs tell me that something has gone wrong?
+    -   validation (did we build the right thing) vs. verification (did we build the thing correctly)
+    -   assertions for sanity checks
+-   How can I ask for help?
+    -   creating a reproducible example (reprex)
+-   How do I install software?
+    -   what *is* a package?
+    -   package manager
+-   What *isn't* included?
+    -   code browsers, multiple cursors, and other fancy IDE tricks
+    -   virtual environments
+    -   debuggers
+
+## Data analysis
+
+-   How can I represent and manage missing values?
+    -   NA
+-   How can I get a feel for my data?
+    -   summary statistics
+-   How can I create a simple model of my data?
+    -   formulas
+    -   linear regression
+        -   adding a best fit straight line on a scatterplot
+        -   understanding what the error bands on a "best fit" straight line mean
+    -   k-means cluster analysis
+    -   frame these as exploratory tools for revealing structure in the data, rather than modelling or inferential tools
+-   How can I put people at risk?
+    -   algorithmic bias
+    -   de-anonymization
+-   What *isn't* included?
+    -   statistical tests
+    -   multiple linear regression
+    -   anything with "machine learning" in its name
+
+## Version control
+
+-   What is a version control system?
+    -   a smarter kind of backup
+-   What goes where and why?
+    -   local vs. remote storage (physically)
+    -   local vs. remote storage (ethical/privacy issues)
+-   How do I track my work locally?
+    -   diff
+    -   add
+    -   commit
+    -   log
+-   How do I view or recover an old version of a file?
+    -   diff
+    -   checkout
+-   How do I save work remotely?
+    -   push and pull
+-   How do I manage conflicts?
+    -   merge
+-   What *isn't* included?
+    -   forking
+    -   branching
+    -   pull requests
```

Comment:
I definitely understand why this won't be included, but from my experience, learners "get it" when they get to collaborate together... but putting this into the course material is difficult and it is a difficult topic. So I'm conflicted on this.

Reply:
They can learn to collaborate by pushing to `master` in a shared repo - that's how we taught Subversion for years. Their cycle becomes `pull`-`merge`-`push`.

Reply:
Can't create a pull request within a single branch - has to be between branches or between repos.

Reply:
Mm, I misunderstood your earlier text. Makes sense.